commit 72ca24bb6484b014ea4d91a323b8a7c8a44085aa
Author: Oliver Henshaw <oliver.henshaw@gmail.com>
Date:   Fri Jan 11 20:04:52 2013 +0000

    Respect Lock boolean in configuration
    
    If not set to require password, don't start the grace period timer and
    always consider in grace period.
    
    REVIEW: 108425
    BUG: 310871
    FIXED-IN: 4.10.1

diff --git a/ksmserver/screenlocker/ksldapp.cpp b/ksmserver/screenlocker/ksldapp.cpp
index d940a95..7f2e671 100644
--- a/ksmserver/screenlocker/ksldapp.cpp
+++ b/ksmserver/screenlocker/ksldapp.cpp
@@ -152,7 +152,11 @@ void KSldApp::configure()
         // timeout stored in seconds
         m_idleId = KIdleTime::instance()->addIdleTimeout(timeout*1000);
     }
-    m_lockGrace = qBound(0, KScreenSaverSettings::lockGrace(), 300000);  // max 5 minutes, keep the value sane
+    if (KScreenSaverSettings::lock()) {
+        m_lockGrace = qBound(0, KScreenSaverSettings::lockGrace(), 300000);  // max 5 minutes, keep the value sane
+    } else {
+        m_lockGrace = -1;
+    }
     m_autoLogoutTimeout = KScreenSaverSettings::autoLogout() ? KScreenSaverSettings::autoLogoutTimeout() * 1000 : 0;
     m_plasmaEnabled = KScreenSaverSettings::plasmaEnabled();
 }
@@ -321,6 +325,8 @@ void KSldApp::idleTimeout(int identifier)
     if (m_lockGrace) {  // short-circuit if grace time is zero
         m_inGraceTime = true;
         m_graceTimer->start(m_lockGrace);
+    } else if (m_lockGrace == -1) {
+        m_inGraceTime = true;  // if no timeout configured, grace time lasts forever
     }
     lock();
 }
